app/back_end/src/tools/spliceai.py

```diff
@@ -1,248 +1,248 @@
+                raise SpliceAIError(f"SpliceAI failed: {result.stderr}")
+            return output_vcf.name  # Return the path to the temporary output VCF file
+        except Exception as exc:
+            raise SpliceAIError(f"Error running SpliceAI: {exc}") from exc
+
+
+def parse_spliceai_vcf(vcf_file: str):
+    """
+    Parses the output VCF file generated by SpliceAI.
+
+    This function extracts the SpliceAI scores from the output VCF file
+    and returns them as a list of dictionaries.
+
+    Args:
+        vcf_file (str): Path to the VCF file with SpliceAI predictions.
+
+    Returns:
+        list: A list of dictionaries containing extracted scores for each variant.
+
+    Raises:
+        SpliceAIError: If there is an error while reading or parsing the VCF file.
+    """
+    spliceai_scores = []
+    try:
+        with open(vcf_file, 'r',encoding='utf-8') as vcf:
+            for line in vcf:
+                if line.startswith('#'):
+                    continue
+                columns = line.strip().split('\t')
+                info_field = columns[7]
+                if "SpliceAI" in info_field:
+                    info_parts = info_field.split(";")
+                    for part in info_parts:
+                        if part.startswith("SpliceAI="):
+                            spliceai_values = part.split("=")[1]
+                            delta_scores = spliceai_values.split("|")
+
+                            ds_ag = float(delta_scores[2])
+                            ds_al = float(delta_scores[3])
+                            ds_dg = float(delta_scores[4])
+                            ds_dl = float(delta_scores[5])
+
+                            max_delta_score = max(ds_ag, ds_al, ds_dg, ds_dl)
+
+                            dp_ag = int(delta_scores[6])
+                            dp_al = int(delta_scores[7])
+                            dp_dg = int(delta_scores[8])
+                            dp_dl = int(delta_scores[9])
+
+                            scores = {
+                                "Delta score (acceptor gain)": ds_ag,
+                                "Delta score (acceptor loss)": ds_al,
+                                "Delta score (donor gain)": ds_dg,
+                                "Delta score (donor loss)": ds_dl,
+                                "Delta position (acceptor gain)": int(columns[1]) + dp_ag,
+                                "Delta position (acceptor loss)": int(columns[1]) + dp_al,
+                                "Delta position (donor gain)": int(columns[1]) + dp_dg,
+                                "Delta position (donor loss)": int(columns[1]) + dp_dl,
+                                "Max_Delta_Score": max_delta_score
+                            }
+                            spliceai_scores.append(scores)
+        return spliceai_scores
+    except Exception as e:
+        raise ValueError(f"Error parsing VCF file {vcf_file}: {e}") from e

```

Comment:
Instead of covering many lines I think it would be better to wrap just one line of usage in next function.